gnome-initial-setup/pages/summary/gis-summary-page.c

```diff
@@ -236,6 +239,117 @@ done_cb (GtkButton *button, GisSummaryPage *page)
+{
+  GisSummaryPage *page = user_data;
+  GisSummaryPagePrivate *priv = gis_summary_page_get_instance_private (page);
+  const gchar *object, *property;
+  g_autoptr(GVariant) value;
+
+  if (g_strcmp0 (signal_name, "ObjectNotify") != 0)
+    return;
+
+  g_variant_get (parameters, "(&s&sv)", &object, &property, &value);
+
+  if (g_strcmp0 (object, "view.JSContext.globalParameters") != 0)
+    return;
+
+  if (g_strcmp0 (property, "unlocked") != 0)
+    return;
+
+  g_return_if_fail (g_variant_is_of_type (value, G_VARIANT_TYPE_BOOLEAN));
+
+  if (g_variant_get_boolean (value))
+    {
+      g_signal_handler_disconnect (priv->clippy_proxy,
+                                   priv->clippy_proxy_signal_id);
+      finish_fbe (page);
+    }
+}
+
+static void
+initial_contact_connect_to_clippy (GisSummaryPage *page)
+{
+  GisSummaryPagePrivate *priv = gis_summary_page_get_instance_private (page);
+  g_autoptr(GError) error = NULL;
+
+  g_dbus_proxy_call_sync (priv->clippy_proxy,
+                          "Connect",
+                          g_variant_new ("(sss)",
+                                         "view.JSContext.globalParameters",
+                                         "notify",
+                                         "unlocked"),
+                          G_DBUS_CALL_FLAGS_NONE,
+                          G_MAXINT,
+                          NULL,
+                          &error);
+
+  if (error != NULL)
+    {
+      g_critical ("Could not call Connect on DBus proxy for HackUnlock App: %s", error->message);
+
+      finish_fbe (page);
+      return;
+    }
+
+  priv->clippy_proxy_signal_id =
+    g_signal_connect (priv->clippy_proxy, "g-signal",
+                      G_CALLBACK (initial_contact_app_signal), page);
+}
+
+static gboolean
+initial_contact_connect_to_clippy_timeout (gpointer data)
+{
+  GisSummaryPage *page = data;
+  initial_contact_connect_to_clippy (page);
+  return G_SOURCE_REMOVE;
+}
+
+static void
+run_initial_contact_app (GisSummaryPage *page)
+{
+  GisSummaryPagePrivate *priv = gis_summary_page_get_instance_private (page);
+  g_autoptr(GError) error = NULL;
+  g_autoptr(GDBusProxy) proxy = NULL;
+
+  proxy = g_dbus_proxy_new_for_bus_sync (G_BUS_TYPE_SESSION,
+                                         G_DBUS_PROXY_FLAGS_NONE,
+                                         NULL,
+                                         "com.endlessm.HackUnlock",
+                                         "/com/endlessm/HackUnlock",
+                                         "com.endlessm.Clippy",
+                                         NULL,
+                                         &error);
+
+  if (error != NULL)
+    {
+      g_critical ("Could not get DBus proxy for HackUnlock App: %s", error->message);
+
+      finish_fbe (page);
+      return;
+    }
+
+  priv->clippy_proxy = g_steal_pointer (&proxy);
+
+  /* FIXME: we need to add a timeout because the Clippy interface is not available yet... */
```

Comment:
This is nasty! Could we at least link to the Phabricator ticket tracking this problem?

Reply:
@wjt There's [this](https://phabricator.endlessm.com/T24047). It can be tracked in it or edited to be more about Clippy and less about the quests stuff.